defaults/main.yml

```diff
@@ -10,16 +10,19 @@ mssql_edition: null
 mssql_tcp_port: 1433
 mssql_manage_firewall: false
 mssql_ip_address: null
-
-mssql_pre_input_sql_file: []
-mssql_post_input_sql_file: []
-mssql_debug: false
-
 mssql_enable_sql_agent: null
 mssql_install_fts: null
 mssql_install_powershell: null
 mssql_enable_ha: null
 mssql_tune_for_fua_storage: null
+mssql_datadir: null
+mssql_datadir_mode: null
```

Comment:
I think you need to have a default value here, either `"0700"` or `"0750"` or  `"0755"` - whatever mssql recommends

Reply:
In this case, the role would overwrite the existing mode back to default if the correct mode is not set with `mssql_datadir_mode`, may be this is expected from Ansible, but I do not think that this is a good UX. I like to be it like this:
If users provide `mssql_datadir_mode`, it is used,
If users don't provide  `mssql_datadir_mode` and the `mssql_datadir` directory exists, the existing mode is kept,
If users don't provide  `mssql_datadir_mode` and the `mssql_datadir` directory doesn't exist, the default mode is used.

There is no recommendation from Microsoft on this, by default the directory is created with `0755`, which is just the default umask on the machine.

Reply:
ok